haystack/utils/import_utils.py

```diff
@@ -5,12 +5,54 @@
 import zipfile
 import requests
 import logging
+import importlib
 from pathlib import Path
 
 
 logger = logging.getLogger(__name__)
 
 
+def safe_import(import_path, classname, dep_group):
```

Comment:
Often there is a comment "optional dependency" after calling this method. Renaming to `import_optional` or `optional_import` might be an idea.

Reply:
Renaming `import_path` to `module_name` might also be an option. Seems to be slightly more precise.

Reply:
Hmm on a second thought: `import_path` is probably slightly more accurate, `module_name` might imply I just need to specify the root module instead of the full path (like `haystack` rather than `haystack.document_stores.elasticsearch`).

I don't like `safe_import` either but `optional_import` might become misleading. The import is going to occur regardless of missing deps, and there's going to be an importable object in the module after calling `safe_import`...